### PR TITLE
[INTER-688] allow overriding runtime endpoints by using worker env

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

**Review by necipallef — CHANGES_REQUESTED** [review]
Could you refactor the code so that the request handler functions take the parameters in the same order? I am talking about below functions:
`handleDownloadScript`
`makeDownloadScriptRequest` (instead of `baseUrl` param, should take `env` param)
`handleIngressAPI`
`makeCacheEndpointRequest` (instead of `ingressBaseUrl` param, should take `env` param, and in the same order)
`makeIngressRequest` (already has `env` param, we can just remove `ingressBaseUrl` param)

**Review by sshelomentsev — DISMISSED** [review]

File: test/endpoints/agentDownload.test.ts [hunk 1]
```diff
@@ -1,13 +1,66 @@
 import worker from '../../src/index'
-import { WorkerEnv } from '../../src/env'
+import { Defaults, WorkerEnv } from '../../src/env'
 import { config } from '../../src/config'
 
 const workerEnv: WorkerEnv = {
+  ...Defaults,
```
Comment (necipallef) [inline test/endpoints/agentDownload.test.ts]: Let's avoid using `Defaults` object in the tests because this object is not related to tests. We can either put `FPJS_CDN_URL=null` and `FPJS_INGRESS_BASE_HOST=null` everywhere, or we can put question marks in front of these fields in the `WorkerEnv` type, like `FPJS_CDN_URL?: string | null`, whichever works for you @TheUnderScorer. This applies to all test files.

**Review by necipallef — CHANGES_REQUESTED** [review]
Let's avoid using `Defaults` object in the tests because this object is not related to tests. We can either put `FPJS_CDN_URL=null` and `FPJS_INGRESS_BASE_HOST=null` everywhere, or we can put question marks in front of these fields in the `WorkerEnv` type, like `FPJS_CDN_URL?: string | null`, whichever works for you @TheUnderScorer. This applies to all test files.

**Comment by necipallef** [comment]
This feature actually makes `src/config.ts` file obsolete. We no more need this file, and we also don't need the `getEnv` function inside the `rollup.config.js`. We can just use the feature this PR introduces, and as a default value, define `fpcdn.io` and `api.fpjs.io` in the `Defaults` object in the `src/env.ts` file. This also means that we change how mocked tests are created in the `teste2e.yml` file (no more need for `env` block there)

Please tell me what you think @TheUnderScorer

**Review by necipallef — APPROVED** [review]
